velox/exec/OutputBuffer.cpp

```diff
@@ -46,8 +46,10 @@ void ArbitraryBuffer::getAvailablePageSizes(std::vector<int64_t>& out) const {
 
 std::vector<std::shared_ptr<SerializedPage>> ArbitraryBuffer::getPages(
     uint64_t maxBytes) {
-  VELOX_CHECK_GT(maxBytes, 0, "maxBytes can't be zero");
-
+  if (maxBytes == 0 && !pages_.empty() && pages_.front() == nullptr) {
```

Comment:
I'm not sure I understand these changes. Would you clarify?

There are more changes in this file re: handling of arbitrary buffer. I don't think I understand these either. Would appreciate some explaining. Thanks.

Reply:
This is to make sure when we call `loadData` with `maxBytes = 0` on a finished and drained arbitrary buffer, we get a end marker.  This situation happens we multiple `DestinationBuffer` polling the same `ArbitraryBuffer`.  With this we can simplify the code in `getData` since we will always get a null marker if there is no data, and we can make sure the callback path there (`sequence - sequence_ >= data_.size()`) always gets notified because the underlying buffer is guaranteed not finished, thus avoid stuck.

Reply:
I see. Would you add comments to the code to clarify this for a future reader? Thanks.